src/Gizmos/scaleGizmo.ts

```diff
@@ -188,6 +236,81 @@ export class ScaleGizmo extends Gizmo {
         return this._sensitivity;
     }
 
+    /**
+     * Builds Gizmo Axis Cache to enable features such as hover state preservation and graying out other axis during manipulation
+     * @param mesh Axis gizmo mesh
+      @param cache display gizmo axis thickness
+     */
+    public addToAxisCache(mesh: Mesh, cache: any) {
+        this.gizmoAxisCache.set(mesh, cache);
+    }
+
+    /**
+     * Subscribes to pointer up, down, and hover events. Used for responsive gizmos.
+     */
+    public subscribeToPointerObserver(): void {
+        const pointerObserver = this.gizmoLayer.utilityLayerScene.onPointerObservable.add((pointerInfo) => {
+            if (pointerInfo.pickInfo) {
+                // On Hover Logic
+                if (pointerInfo.type === PointerEventTypes.POINTERMOVE) {
+                    if (this.dragging) { return; }
+                    this.gizmoAxisCache.forEach((statusMap, parentMesh) => {
+                        const isHovered = pointerInfo.pickInfo && (parentMesh.getChildMeshes().indexOf((pointerInfo.pickInfo.pickedMesh as Mesh)) != -1);

```

Comment:
Also - no need to test for pickInfo again. if typescript gives you a hard time, use the `!` character after pickInfo (of course only in this case, as you are chekcing it before)